rust/agama-software/src/callbacks.rs

```diff
@@ -9,11 +9,22 @@ mod security;
 pub use security::Security;
 mod install;
 pub use install::Install;
-use tokio::runtime::Handle;
+use std::sync::mpsc;
+use tokio::runtime::Builder;
 
 fn ask_software_question(
     handler: &Handler<question::Service>,
     question: QuestionSpec,
 ) -> Result<Answer, AskError> {
-    Handle::current().block_on(async move { ask_question(handler, question).await })
+    let (tx, rx) = mpsc::channel();
+    let question_handler = handler.clone();
+    std::thread::spawn(move || {
+        let runtime = Builder::new_current_thread().enable_all().build().unwrap();
+        let result =
+            runtime.block_on(async move { ask_question(&question_handler, question).await });
+        // unwrap OK: the receiver should not be dropped before the thread finishes
+        tx.send(result).unwrap();
+    });
+    // unwrap OK: the sender should not be dropped before sending a value
+    rx.recv().unwrap()
```

Comment:
I worry that this won't work as recv is async method https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Receiver.html#method.recv
I am curious why rust do not complain that you do not call async.

Reply:
yeap, that is part which  I am not sure. For now we use spawn_blocking on libzypp thread, but I am not sure if other part of channel do not block whole runtime of actor.